jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -220,21 +197,41 @@ def _on_viewer_data_changed(self, msg):
         else:
             self.disabled_msg = ''
 
+    def _reset_results_and_marks(self):
+        """Reset results, results_available, and clear continuum marks."""
+        self.results_available = False
+        self.results = [{'function': function, 'result': ''} for function in FUNCTIONS]
+        self._update_continuum_marks()
+
     def _on_viewer_subsets_changed(self, msg):
         """
-        Update the statistics if any of the referenced regions have changed
+        Update the statistics if the current selection spectral region has been
+        modified or deleted.
 
         Parameters
         ----------
         msg : `glue.core.Message`
             The glue message passed to this callback method.
         """
-        if (msg.subset.label in [self.spectral_subset_selected,
-                                 self.continuum_subset_selected]):
-            self._calculate_statistics(msg)
+
+        current_selections = [self.spectral_subset_selected,
+                              self.continuum_subset_selected]
+
+        # If a currently selected subset is deleted, just clear marks and statistics.
+        # The re-calculation of statistics will happen subsequently when the default
+        # selection is applied in the drop down and this method is called again
+        if isinstance(msg, SubsetDeleteMessage):
+            if msg.subset.label in current_selections:
+                self._reset_results_and_marks()
+                return
+
+        else:
+            if msg.subset.label in current_selections:
+                self._calculate_statistics(msg)
 
     def _on_global_display_unit_changed(self, msg):
-        self._calculate_statistics(msg)
+        update_marks = msg.axis != 'spectral'
+        self._calculate_statistics(msg, update_marks=update_marks)
```

Comment:
this is because otherwise the unit conversion is happening twice?  Is that something we can improve in the marks logic itself (perhaps we need to also add support in marks for handling flux unit conversion) and avoid this switch?